qcore/siteamp_models.py

```diff
@@ -58,107 +52,553 @@ def amplification_uncertainty(
 
-    # fmt: off
-    freqs = 1.0 / np.array([0.001, 0.01, 0.02, 0.03, 0.05, 0.075, 0.10,
-                            0.15, 0.20, 0.25, 0.30, 0.40, 0.50, 0.75,
-                            1.00, 1.50, 2.00, 3.00, 4.00, 5.00, 7.50, 10.0])
-    if version == '2008':
-        c10 = np.array([1.058, 1.058, 1.102, 1.174, 1.272, 1.438, 1.604,
-                        1.928, 2.194, 2.351, 2.460, 2.587, 2.544, 2.133,
-                        1.571, 0.406,-0.456,-0.82, -0.82, -0.82, -0.82, -0.82])
-    elif version == '2014':
-        # named c11 in cb2014
-        c10 = np.array([1.090, 1.094, 1.149, 1.290, 1.449, 1.535, 1.615,
-                        1.877, 2.069, 2.205, 2.306, 2.398, 2.355, 1.995,
-                        1.447, 0.330, -0.514, -0.848, -0.793, -0.748, -0.664,
-                        -0.576])
+
+@njit(cache=True)
+def _compute_fs_value(
+    t_idx: int,
+    vs30: float,
+    a1100: np.ndarray,
+    c10: np.ndarray,
+    k1: np.ndarray,
+    k2: np.ndarray,
+):
+    """Compute site factor based on vs30 value
+
+    Parameters
+    ----------
+    t_idx : int
+        The index to compute for.
+    vs30 : float
+        The reference vs30
+    a1100, c10, k1, k2 : np.ndarray
+        Parameters for calculation
+
+    Returns
+    -------
+    np.ndarray
+         Site amplification factor.
+    """
+    if vs30 < k1[t_idx]:
+        return _fs_low(t_idx, vs30, a1100, c10, k1, k2)
+    elif vs30 < 1100.0:
+        return _fs_mid(t_idx, vs30, c10, k1, k2)
     else:
-        raise Exception(f"BAD CB AMP version specified: {version}")
-    k1 = np.array([865.0, 865.0, 865.0, 908.0, 1054.0, 1086.0, 1032.0,
-                   878.0, 748.0, 654.0, 587.0,  503.0,  457.0,  410.0,
-                   400.0, 400.0, 400.0, 400.0,  400.0,  400.0,  400.0, 400.0])
-    k2 = np.array([-1.186, -1.186, -1.219, -1.273, -1.346, -1.471, -1.624,
-                   -1.931, -2.188, -2.381, -2.518, -2.657, -2.669, -2.401,
-                   -1.955, -1.025, -0.299,  0.0,    0.0,    0.0,    0.0, 0.0])
-    # fmt: on
-
-    # f_site function domains
-    def fs_low(T, vs30, a1100):
-        return c10[T] * log(vs30 / k1[T]) + k2[T] * log(
-            (a1100 + scon_c * exp(scon_n * log(vs30 / k1[T]))) / (a1100 + scon_c)
-        )
+        return _fs_high(t_idx, c10, k1, k2)
+
+
+AMPLIFICATION_FREQUENCIES = 1.0 / np.array(
+    [
+        0.001,
+        0.01,
+        0.02,
+        0.03,
+        0.05,
+        0.075,
+        0.10,
+        0.15,
+        0.20,
+        0.25,
+        0.30,
+        0.40,
+        0.50,
+        0.75,
+        1.00,
+        1.50,
+        2.00,
+        3.00,
+        4.00,
+        5.00,
+        7.50,
+        10.0,
+    ]
+)
+
+
+@njit(cache=True)
+def _cb_amp(
+    vref: float,
+    vsite: float,
+    vpga: float,
+    pga: float,
+    version: int = 2014,  # Changed to integer
+    flowcap: float = 0.0,
+    freqs: np.ndarray = AMPLIFICATION_FREQUENCIES,
+) -> np.ndarray:
+    """
+    Numba translation of cb_amp.
```

Comment:
Do we have a function that explain what cb_amp is at all? Might be better for at least a short description with the addition of numba translation?